src/frontend/src/optimizer/plan_node/logical_source.rs

```diff
@@ -114,13 +161,188 @@ impl ColPrunable for LogicalSource {
     }
 }
 
+/// A util function to extract kafka offset timestamp range.
+///
+/// Currently we only support limiting kafka offset timestamp range using literals, e.g. we only
+/// support expressions like `_rw_kafka_timestamp <= '2022-10-11 1:00:00+00:00'`.
+///
+/// # Parameters
+///
+/// * `expr`: Expression to be consumed.
+/// * `range`: Original timestamp range, if `expr` can be recognized, we will update `range`.
+/// * `schema`: Input schema.
+///
+/// # Return Value
+///
+/// If `expr` can be recognized and consumed by this function, then we return `None`.
+/// Otherwise `expr` is returned.
+fn expr_to_kafka_timestamp_range(
```

Comment:
I don't get why we still have a time range in the optimizer. Is not all splits that don't match time requirements are dismissed?

Reply:
Timestamp range will apply to every split.

Reply:
We will extract the timestamp literal in planner or optimizer. And use kafka_enumerator.list_splits_batch in plan_fragmet to covert timestamp to the according offset. 

BTW, this pr don't filter the invalid offset range. As we discuss in https://github.com/risingwavelabs/risingwave/pull/7112#discussion_r1059224681_, directly return empty vec if reader find the offset range is empty is good enough.